test/auth/authentication-client.tests.js

```diff
@@ -67,8 +68,73 @@ describe('AuthenticationClient', function() {
     }
   });
 
+  describe('client info', function() {
+    it('should generate default telemetry value', function() {
+      var client = new AuthenticationClient({ token: 'token', domain: 'auth0.com' });
+      var pkgVersion = require('../../package.json').version;
+      var nodeVersion = process.version.replace('v', '');
+      expect(client.getClientInfo()).to.deep.equal({
+        name: 'node-auth0',
+        version: pkgVersion,
+        env: [{ node: nodeVersion }]
+      });
+    });
+
+    it('should configure instances with default telemetry header', function() {
+      sinon.stub(AuthenticationClient.prototype, 'getClientInfo', function() {
+        return { name: 'test-sdk', version: 'ver-123' };
+      });
+      var client = new AuthenticationClient({ token: 'token', domain: 'auth0.com' });
+
+      var requestHeaders = {
+        'Auth0-Client': 'eyJuYW1lIjoidGVzdC1zZGsiLCJ2ZXJzaW9uIjoidmVyLTEyMyJ9',
+        'Content-Type': 'application/json'
+      };
+      expect(client.oauth.oauth.options.headers).to.contain(requestHeaders);
+      expect(client.database.dbConnections.options.headers).to.contain(requestHeaders);
+      expect(client.passwordless.passwordless.options.headers).to.contain(requestHeaders);
+      expect(client.users.headers).to.contain(requestHeaders);
+      expect(client.tokens.headers).to.contain(requestHeaders);
+
+      AuthenticationClient.prototype.getClientInfo.restore();
+    });
+
+    it('should configure instances with custom telemetry header', function() {
+      var customTelemetry = { name: 'custom', version: 'beta-01', env: { node: 'v10' } };
```

Comment:
Not to open a can of worms here ... but someone can just pass whatever and it goes through? What about telemetry without a `name`?

Reply:
Yes. They can. I think the decision for other SDKs was to prevent the telemetry being sent at all if the name was not present, should I do that here as well?

Reply:
@lbalmaceda - Seems like a good idea to me. Then at least we can filter _some_ of the garbage that might be sent here.